system/HTTP/CURLRequest.php

```diff
@@ -649,6 +649,17 @@ protected function setCURLOptions(array $curlOptions = [], array $config = [])
             $this->setHeader('Content-Length', (string) strlen($json));
         }
 
+        // Resolve IP
+        if (array_key_exists('force_ip_resolve', $config) && $config['force_ip_resolve']) {
+            $protocolVersion = $config['force_ip_resolve'];
+
+            if ($protocolVersion === 'v4') {
+                $curlOptions[CURLOPT_IPRESOLVE] = CURL_IPRESOLVE_V4;
+            } elseif ($protocolVersion === 'v6') {
+                $curlOptions[CURLOPT_IPRESOLVE] = CURL_IPRESOLVE_V6;
+            }
+        }
```

Comment:
How about:

```suggestion
        if (array_key_exists('force_ip_resolve', $config)) {
            $curlOptions[CURLOPT_IPRESOLVE] = match($config['force_ip_resolve']) {
                'v4' => CURL_IPRESOLVE_V4,
                'v6' => CURL_IPRESOLVE_V6,
                default => CURL_IPRESOLVE_WHATEVER
            };
        }
```

Reply:
or throw for unknown value?

Reply:
Hmm... yes, that's probably even better than silently selecting the default value.

Reply:
I think dont any insert option when no call

Reply:
This options same behaviour in `version`. If there no set declaration, we follow default of PHP Curl

Reply:
What if I initialize this service with custom options? I can't change `CURLOPT_IPRESOLVE` to the default value later?

This seems like an issue - same story with `version`. There is no option to reset it to the default value during the instance lifecycle.